eventloop/eventloop.go

```diff
@@ -27,15 +27,32 @@ func GetGlobalEventLoop() *EventLoop {
 func (e *EventLoop) Await(currentP *Promise) interface{} {
 	defer currentP.Done()
 	currentP.RegisterHandler()
-	val := <-currentP.rev
-	return val
+	for {
+		select {
+		case err := <-currentP.errChan:
+			return err
+		case rev := <-currentP.rev:
+			return rev
+		}
+	}
+
 }
 
 func (e *EventLoop) Async(fn func() (interface{}, error)) *Promise {
 	resultChan := make(chan interface{})
 	errChan := make(chan error)
 	p := e.newPromise(resultChan, errChan)
 	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				switch x := r.(type) {
+				case error:
+					p.errChan <- x
+				default:
+					p.errChan <- fmt.Errorf(`unknown error: %v`, x)
+				}
+			}
+		}()
```

Comment:
I like this idea, but now we are banking on the user always implementing a catch block. if for example a user doesn't this code will block forever. We need to think about how we would handle this.

Reply:
True, this will need some deliberation 😄